src/Aspire.Hosting.Azure/AzureResourceExtensions.cs

```diff
@@ -309,4 +309,29 @@ private static void WriteAzureOpenAIDeploymentToManifest(ManifestPublishingConte
         context.Writer.WriteString("type", "azure.openai.deployment.v0");
         context.Writer.WriteString("parent", resource.Parent.Name);
     }
+
+    /// <summary>
+    /// Adds an Azure Application Insights resource to the application model.
+    /// </summary>
+    /// <param name="serverBuilder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="connectionString">The connection string.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{AzureSqlDatabaseResource}"/>.</returns>
+    public static IResourceBuilder<AzureApplicationInsightsResource> AddApplicationInsights(
+        this IDistributedApplicationBuilder serverBuilder,
+        string name,
+        string? connectionString = null)
+    {
+        var appInsights = new AzureApplicationInsightsResource(name, connectionString);
+        return serverBuilder.AddResource(appInsights)
+                        .WithManifestPublishingCallback(WriteAzureApplicationInsightsDeploymentToManifest);
+    }
+
```

Comment:
We also need to overload WithReference to set the right environment variable:

Copy this https://github.com/dotnet/aspire/blob/9a591b349ebffee20862b5ed281c1244b9c7eeec/src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs#L149-L186 and change it so that it sets APPLICATIONINSIGHTS_CONNECTION_STRING.

See https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/application-insights for manual mode today 😄.